go/kbfs/libkbfs/subscription_manager.go

```diff
@@ -94,6 +94,109 @@ type pathSubscriptionRef struct {
 	path         cleanInTlfPath
 }
 
+type onlineStatusTracker struct {
+	cancel   func()
+	config   Config
+	onChange func()
+
+	lock          sync.RWMutex
+	currentStatus keybase1.KbfsOnlineStatus
+}
+
+const onlineStatusTryingStateTimeout = 4 * time.Second
+
+func (ost *onlineStatusTracker) updateOnlineStatus(
+	currentStatus keybase1.KbfsOnlineStatus) {
+	ost.lock.Lock()
+	defer ost.lock.Unlock()
+	if ost.currentStatus == currentStatus {
+		return
+	}
+	ost.currentStatus = currentStatus
+	ost.onChange()
+}
+
+func (ost *onlineStatusTracker) updateOnlineStatusIfCurrentIs(
+	expected keybase1.KbfsOnlineStatus,
+	currentStatus keybase1.KbfsOnlineStatus) (updated bool) {
+	ost.lock.Lock()
+	defer ost.lock.Unlock()
+	if ost.currentStatus == expected {
+		ost.currentStatus = currentStatus
+		ost.onChange()
+		return true
+	}
+	return false
+}
+
+// GetOnlineStatus implements the OnlineStatusTracker interface.
+func (ost *onlineStatusTracker) GetOnlineStatus() keybase1.KbfsOnlineStatus {
+	ost.lock.RLock()
+	defer ost.lock.RUnlock()
+	return ost.currentStatus
+}
+
+func (ost *onlineStatusTracker) myWatchBegins(ctx context.Context) {
+	go func() {
```

Comment:
You can avoid an indent in this function by doing the `go` in the caller instead.